backend/src/main/java/com/google/sps/FindMatchQuery.java

```diff
@@ -16,71 +16,65 @@
+  private static final int PADDING_TIME = 10;
+  /** Conversion from minutes to milliseconds */
+  private static final int MINUTES_TO_MILLIS = 60000;
   /** Reference clock */
-  private Clock clock;
+  private final Clock clock;
+  /** Datastore of Participants */
+  private final ParticipantDatastore participantDatastore;
 
   /** Constructor */
-  public FindMatchQuery(Clock clock) {
+  public FindMatchQuery(Clock clock, ParticipantDatastore participantDatastore) {
     this.clock = clock;
+    this.participantDatastore = participantDatastore;
   }
 
   /**
-   * Find match of new participant or add to list of participants, return whether or not match was
-   * found right after being added
+   * @return Match of new participant with unmatched participants by comparing duration and
+   *     availibility, or null if no match yet
    */
-  public Match findMatch(List<Participant> participants, Participant newParticipant) {
-    // Set reference date time using clock
-    ZonedDateTime dateTime = ZonedDateTime.now(clock);
-    System.out.println(dateTime);
+  @Nullable
+  public Match findMatch(Participant newParticipant) {
+    int duration = newParticipant.getDuration();
 
-    // Compare new participant preferences with others in list to find match
-    for (Participant currParticipant : participants) {
+    // Get list of unmatched participants with same duration as newParticipant
+    List<Participant> sameDurationParticipants =
+        participantDatastore.getParticipantsWithDuration(duration);
 
-      // Check if participants are looking for similar meeting duration
-      int newDuration = newParticipant.getDuration();
-      int currDuration = currParticipant.getDuration();
-      boolean compatibleDuration = Math.abs(newDuration - currDuration) <= MAX_DURATION_DIFF;
+    // Set reference date time using clock
+    long currentTimeMillis = clock.millis();
 
-      if (!compatibleDuration) {
-        continue;
-      }
-      int duration = Math.min(newDuration, currDuration);
+    long newEndTimeAvailable = newParticipant.getEndTimeAvailable();
 
+    // Compare new participant preferences with other participants to find match
+    for (Participant currParticipant : sameDurationParticipants) {
       // Check if participants are both free for that duration + extra
-      ZonedDateTime newEndTimeAvailable = newParticipant.getEndTimeAvailable();
-      ZonedDateTime currEndTimeAvailable = currParticipant.getEndTimeAvailable();
-      ZonedDateTime earliestEndTimeAvailable =
-          getEarlier(newEndTimeAvailable, currEndTimeAvailable);
+      long currEndTimeAvailable = currParticipant.getEndTimeAvailable();
+      long earliestEndTimeAvailable = Math.min(newEndTimeAvailable, currEndTimeAvailable);
       boolean compatibleTime =
-          dateTime.plusMinutes(duration + PADDING_TIME).isBefore(earliestEndTimeAvailable);
+          (currentTimeMillis + (duration + PADDING_TIME) * MINUTES_TO_MILLIS)
```

Comment:
TimeUnit.MINUTES.toMillis(duration + PADDING_TIME))

Also I think it would be best to document that PADDING_TIME is in minutes, so maybe `PADDING_TIME_MINS` or `PADDING_TIME_MINUTES`